### PR TITLE
Fix #7781: Avoid using nested package definition

### LINKED ISSUE #7781: Check bootstrapping issue with `quoted.Expr`
Check bootstrapping issue with `quoted.Expr`.

### PR DESCRIPTION
**Pull request by nicolasstucki**
Had to remove `sealed` from `Expr` and `Type`

### CODE REVIEW

**Comment by odersky** [comment]
I think I have an alternative fix for that

**Comment by odersky** [comment]
With #7786 this change should be no longer necessary.

**Comment by odersky** [comment]
Reopened since we won't go ahead with #7786.

File: tests/patmat/i6255b.check [hunk 1]
```diff
@@ -0,0 +1 @@
+2: Pattern Match Exhaustivity: _: Expr[Int]
```
Comment (liufengyun) [inline tests/patmat/i6255b.check]: This is interesting, we don't have the warning before. I'll have a look at it.

Comment (nicolasstucki) [reply tests/patmat/i6255b.check]: It made sense because Expr is not sealed anymore. The test case was clearly not exhaustive.

Comment (nicolasstucki) [reply tests/patmat/i6255b.check]: I wonder why the error was not there before

**Pull request merged**